robot/basestation/app.py

```diff
@@ -205,54 +204,61 @@ def rover_drive():
 
     return jsonify(success=True, cmd=cmd, feedback=feedback, error=error)
 
-# capture image
-@app.route("/capture_image", methods=["POST", "GET"])
+@app.route("/capture_image/", methods=["POST", "GET"])
 def capture_image():
-    stream_url = "http://" + fetch_ros_master_ip() + ":8080/stream?topic=/cv_camera/image_raw"
-    #lserror, lsoutput = run_shell("ls -1q img* | wc -l")
-    # p1 = subprocess.Popen(split("ls -1q img*"), stdout=subprocess.PIPE)
-    # p2 = subprocess.Popen(split("wc -l"), stdin=p1.stdout)
-    # output, error = p2.communicate()
-    output, error = run_shell('ls')
-    output = output.decode()
-    print('output:', output)
-    i = 0
-
-    if 'img' in output:
-        i = output.rfind('img')
-        i = int(output[i + 3]) + 1 # shift by 'img'
-        print('i', i)
-
-    error, output = run_shell("ffmpeg -i " + stream_url + " -ss 00:00:01.500 -f image2 -vframes 1 img" + str(i) + ".jpg")
-    msg = "success"
+    """ Given a stream, captures an image.
 
-    if error:
-        msg = "F"
+        stream_url : The URL of the stream to capture the image.
+    """
+    stream_url = request.args['stream_url']
+    image_directory = "stream_images"
```

Comment:
Shouldn't there be a directories with the shorthand stream names, in order to separate the images taken from different cameras? Like how the stream video clips are saved.

Reply:
good idea. since we already use this for stream recording I will do the same